models/classes/runner/TestDefinitionSerializerService.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+namespace oat\taoQtiTest\models\runner;
+
+use core_kernel_classes_Literal;
+use core_kernel_classes_Property;
+use oat\oatbox\service\ConfigurableService;
+use tao_models_classes_service_FileStorage;
+use taoQtiTest_models_classes_QtiTestService;
+
+class TestDefinitionSerializerService extends ConfigurableService
+{
+    const SERVICE_ID = 'taoQtiTest/TestDefinitionSerializerService';
+
+    /**
+     * @param QtiRunnerServiceContext $serviceContext
+     * @return array
+     * @throws \common_exception_Error
+     * @throws \common_exception_InconsistentData
+     * @throws \core_kernel_persistence_Exception
+     * @throws \oat\tao\model\websource\WebsourceNotFound
+     */
+    public function getSerializedTestDefinition(QtiRunnerServiceContext $serviceContext)
+    {
+        return $this->parseXmlToArray($this->getTestDefinitionFilePath($serviceContext));
+    }
+
+    /**
+     * @param string $filePath
+     * @return array
+     */
+    private function parseXmlToArray($filePath)
+    {
+        $xml = simplexml_load_file($filePath);
+        $json = json_decode(json_encode($xml), true);
```

Comment:
This is an associative array, not a json

Reply:
Fixed.